cdap-api/src/main/java/co/cask/cdap/api/macro/MacroEvaluator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.macro;
+
+import co.cask.cdap.api.annotation.Beta;
+
+/**
+ * Macro evaluator
+ */
+@Beta
+public interface MacroEvaluator {
+
+  /**
+   * lookup the property and return the value corresponding to the property.
+   */
+  String evaluate(String property);
+
+  /**
+   * Use the macro function and call the function with provided arguments,
+   * function uses the arguments and returns the evaluated response.
```

Comment:
InvalidMacroException is thrown during configure time when we check for syntax. Its possible the macro evaluates to another invalid macro which can cause InvalidMacroException. so ill add that. macro function can cause other runtime exception, should we use the same InvalidMacroException for them?